compute_endpoint/tests/unit/test_endpointmanager_unit.py

```diff
@@ -267,6 +278,36 @@ def test_gracefully_exits_if_registration_blocked(
         assert pyexc.value.http_status == status_code
 
 
+def test_handles_provided_endpoint_id_no_json(
+    mock_client: t.Tuple[uuid.UUID, mock.Mock],
+    conf_dir: pathlib.Path,
+    mock_conf: Config,
+):
+    ep_uuid, mock_gcc = mock_client
+
+    EndpointManager(conf_dir, ep_uuid, mock_conf)
+
+    _a, k = mock_gcc.register_endpoint.call_args
+    assert k["endpoint_id"] == ep_uuid
+
+
+def test_handles_provided_endpoint_id_with_json(
+    mock_client: t.Tuple[uuid.UUID, mock.Mock],
+    conf_dir: pathlib.Path,
+    mock_conf: Config,
+):
+    ep_uuid, mock_gcc = mock_client
+    provided_ep_uuid_str = str(uuid.uuid4())
+
+    ep_json = conf_dir / "endpoint.json"
+    ep_json.write_text(json.dumps({"endpoint_id": str(ep_uuid)}))
+
+    EndpointManager(conf_dir, provided_ep_uuid_str, mock_conf)
+
+    _a, k = mock_gcc.register_endpoint.call_args
+    assert k["endpoint_id"] == ep_uuid
```

Comment:
Just making sure I follow what this test is saying ... that the data in `endpoint.json` overrides whatever might be passed on the command line?

Reply:
That's correct; this matches the previous behavior.